drivers/dac/ltc2672/ltc2672.c

```diff
@@ -139,39 +139,125 @@ int ltc2672_transaction(struct ltc2672_dev *device, uint32_t comm, bool is_32)
+ */
+float ltc2672_code_to_current(struct ltc2672_dev *device, uint16_t dac_code,
+			      enum ltc2672_dac_ch out_ch)
+{
+	float dac_current;
+
+	if (device->id == LTC2672_12)
+		dac_current = ((float)dac_code/LTC2672_12BIT_RESO)*device->max_currents[out_ch];
+	else
+		dac_current = ((float)dac_code/LTC2672_16BIT_RESO)*device->max_currents[out_ch];
+
+	return dac_current;
+}
+
+/**
+ * @brief sets the dac code for channel
+ * @param device - ltc2672 descriptor
+ * @param code - The code to be written
+ * @param out_ch - channel to set the code
  * @return 0 in case of success, negative error code otherwise
  */
-int ltc2672_set_current_channel(struct ltc2672_dev *device, float current,
-				enum ltc2672_dac_ch out_ch)
+int ltc2672_set_code_channel(struct ltc2672_dev *device, uint16_t code,
+			     enum ltc2672_dac_ch out_ch)
 {
-	float current_code;
 	uint32_t command;
 
 	/* Switching to V- results in constant -80mA output */
 	if (device->out_spans[out_ch] == LTC2672_VMINUS_VREF) {
-		command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_PWRUP_UPD_CHANNEL_X, out_ch,
+		command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_PWRUP_UPD_CHANNEL_X,
+						     out_ch,
 						     LTC2672_DUMMY);
 
 		return ltc2672_transaction(device, command, false);
 	}
 
+	if (device->id == LTC2672_12)
+		code <<= 4;
+
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_PWRUP_UPD_CHANNEL_X,
+					     out_ch,
+					     code);
+
+	return ltc2672_transaction(device, command, false);
+}
+
+/**
+ * @brief set the current for a selected DAC channel
+ * @param device - ltc2672 descriptor
+ * @param current - current to set in mA
+ * @param out_ch - channel to set the current
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_set_current_channel(struct ltc2672_dev *device,
+				float current,
+				enum ltc2672_dac_ch out_ch)
+{
+	uint32_t current_code;
+
 	if (current > device->max_currents[out_ch] || current < LTC2672_OFF_CURRENT
 	    || out_ch < LTC2672_DAC0 || out_ch > LTC2672_DAC4)
 		return -EINVAL;
 
-	current_code = (float)(current / device->max_currents[out_ch]);
+	current_code = ltc2672_current_to_code(device, current, out_ch);
+
+	return ltc2672_set_code_channel(device, current_code, out_ch);
+}
+
+/**
+ * @brief sets the same dac code for all channels
+ * @param device - ltc2672 descriptor
+ * @param code - The code to be written
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_set_code_all_channels(struct ltc2672_dev *device, uint16_t code)
+{
+	uint32_t command;
+
+	if (device->out_spans[0] == LTC2672_VMINUS_VREF) {
+		command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_PWRUP_UPD_CHANNEL_ALL,
+						     LTC2672_DAC0,
+						     LTC2672_DUMMY);
+
+		return ltc2672_transaction(device, command, false);
+	}
 
 	if (device->id == LTC2672_12)
-		current_code = (uint32_t)(current_code * (float)(LTC2672_12BIT_RESO)) << 4;
-	else
-		current_code *= LTC2672_16BIT_RESO;
+		code <<= 4;
```

Comment:
what does this shift do? please use relevant macros.

Reply:
Defined a macro for shifting required for a 12bit resolution type device for appropriate command creation